DEVELOPER_GUIDE.md

```diff
@@ -494,3 +497,15 @@ Pass a list of files or directories to limit your search.
 The Github workflow in [lucene-snapshots.yml](.github/workflows/lucene-snapshots.yml) is a Github worfklow executable by maintainers to build a top-down snapshot build of lucene.
 These snapshots are available to test compatibility with upcoming changes to Lucene by updating the version at [version.properties](buildsrc/version.properties) with the `version-snapshot-sha` version.
 Example: `lucene = 10.0.0-snapshot-2e941fc`.
+
+# Creating uber-JAR/fat-JAR of a module
+An uber-JAR is the JAR, which contains classes from all the libraries, on which your project depends and, of course, the classes of current project.
+
+There might be cases where the developer would like to add some custom logic to the code of a module (like the rest client) and generate anuber-JAR that can be directly used by the dependency management tool.
```

Comment:
For example, in https://github.com/opensearch-project/OpenSearch/issues/3702 a developer needed ...